### PR TITLE
V2 Fix interface nil checks

### PR DESCRIPTION
**Pull request by jwhonce**
Bugs fixes:

* Accept connections from services that don't provide version header
* Fix interface parameter nil checks. See https://golang.org/doc/faq#nil_error

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
your signing is off ... fix and we can review and merge

**Comment by baude** [comment]
lgtm

File: pkg/bindings/containers/logs.go [hunk 1]
```diff
@@ -50,7 +50,6 @@ func Logs(ctx context.Context, nameOrID string, opts LogOptions, stdoutChan, std
 	if err != nil {
 		return err
 	}
-	defer response.Body.Close()
```
Comment (mheon) [inline pkg/bindings/containers/logs.go:53]: Why was this removed?

**Comment by mheon** [comment]
One question otherwise LGTM

File: pkg/bindings/containers/logs.go (hunk 1 above)
Comment (jwhonce) [reply pkg/bindings/containers/logs.go:53]: @mheon It could cause a double close on the body.
